### ISSUE TITLE
New quests  for staircases

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#503]
Does this [staircase](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dsteps) has:
* a [handrail](https://wiki.openstreetmap.org/wiki/Key:handrail)? (possibly, if so, where: right or left or both)
* and how many [steps it has](https://wiki.openstreetmap.org/wiki/Key:step_count) (**Edit:** tracked in https://github.com/westnordost/StreetComplete/issues/875)
* for wheelchair users and bicycles, maybe again ask for [ramp](https://wiki.openstreetmap.org/wiki/Key:ramp)?
  I know there are few ramps considering all staircases, but as the note from wiki says: "It's equally important to use ramp=no."

This information may useful for people with difficulties passing staircases, blind people, :biking_man:/:biking_woman:…

**Comment by matkoniecz** [comment]
Maybe split into a separate issues? 

"and how many steps it has" seems to be ridiculous, requires significant efforts for longer ones and give extremely limited benefits (is there even single case of using this data?).

**Comment by westnordost** [comment]
Who is going to look on his (smartphone) map to find if a staircase has a handrail (and then avoid this staircase)? To me, it sounds like super-unimportant information. Do you know _any_ data consumer that uses this?
Same with steps. Also, steps can be inferred roughly from the length of the way.
Ramp might be slightly helpful for routing applications.

**Comment by matkoniecz** [comment]
> to find if a staircase has a handrail

In theory it may be used for routing and 3D rendering. I am not aware about anybody who uses or wants to use this data.

> ramp

Note "Ramps that run physically separated (e.g. separated by a fence, height difference or wall), should be drawn separately.". There is no defined key to indicate this (maybe ramp=no should be used in such cases? it is not clear).

**Comment by rugk** [comment]
> "and how many steps it has" seems to be ridiculous […] nd give extremely limited benefits (is there even single case of using this data?).

Not at all. See the [wiki page](https://wiki.openstreetmap.org/wiki/Key:step_count#Use_cases). It actually lists **5 use cases** (blind people, bicyclists, 3d rendering, …) and never forget what other use cases might come up when one has enough data in the future…

> requires significant efforts for longer ones

Indeed it does, but nobody forces the StreetComplete user to solve that quest. In any case, I agree, however, that the step_count quest should have the lowest priority of all staircase quests.

> Maybe split into a separate issues?

Yeah, could do so, but I thought these 2 or 3 quests are so closely related that it is okay to have one issue for that.

**Comment by rugk** [comment]
> Do you know any data consumer that uses this?

See above. Actually I've requested way less important quests here and this is IMHO already a good thing. Of course you are unlikely to consume this data manually, but routing engines, 3d models and other stuff can take care of it.

**Comment by rugk** [comment]
BTW: Do you never read the wiki page for a quest suggestion before complaining about it? There are exact examples, so hey…

**Comment by matkoniecz** [comment]
> 5 use cases

I was asking about whatever somebody is **using** this data. This question is not answered on any linked page from what I see.

My reaction may be influenced by fact that I recently walked through several highway=steps way, each with 500+ plus steps (though maybe asking for very short ways may be feasible...).

**Comment by rugk** [comment]
Okay 500+ steps is crazy (when creating this quest, I mostly thought of some small 5-10 steps, where you then know you can just lift your bike up), but walking them, I think, you can count them quite easily. I think there are some people always doing it when walking staircases. :wink: 
In any case easier to count than certain parking areas, maybe… (https://github.com/westnordost/StreetComplete/issues/502) :smile: 

> I was asking about whatever somebody is using this data.

Still don't understand what you mean.

**Comment by westnordost** [comment]
> See above. Actually I've requested way less important quests here

This is not an argument (in your favour). I'd rather not spend time on discussing whether or not to include those borderline next-to-irrelevant quests while there are other more important quests to do, really.

**Comment by westnordost** [comment]
There is no way in the github issue tracker to properly prioritise these things, so I am swamped with a lot of these kind of quest suggestions that block the sight to the actually important ones (such as the building type quest) and give possible contributors the wrong idea about which should be the priorities.
When a PR comes, I don't want to outright close it just because other quests should be done first. So, I need to review these PRs, comment, request changes, review again and finally pull and/or even help out (i.e. make an icon etc.)

This all is a huge waste of time.

**Comment by rugk** [comment]
Then leave it open for PRs. This is open-source and also a community project… That's the big advantage of FLOSS software.

**Comment by rugk** [comment]
> There is no way in the github issue tracker to properly prioritise these things, 

Some projects create tags: "priority:P1", and so on… Maybe do this?

Also I can suggest this: https://github.com/westnordost/StreetComplete/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by westnordost** [comment]
Well, it's not about whether **I** can keep a prioritised overview over the issue tracker but whether contributors can. So this would need to be the default view.

Anyway, enough of my whining.

For this, as said, `ramp` as a quest is fine for now.

**Comment by westnordost** [comment]
http://devopsreactions.tumblr.com/post/146398379161/when-the-tickets-just-keep-coming

**Comment by westnordost** [comment]
> Really, actually I think this is the property here, which is the most difficult/dubious whether it can be implemented in a good way. See #503 (comment).

Perhaps the wheelchair-ramp. But the others are fine, IMO

**Comment by matkoniecz** [comment]
> I can keep a prioritised overview over the issue tracker but whether contributors can

I have seen some projects with "very low importance" tags like https://github.com/MarcusWolschon/osmeditor4android/issues?q=is%3Aissue+is%3Aopen+label%3A%22Low+Priority%22

**Comment by rugk** [comment]
Exactly, that's what I mean.

**Comment by westnordost** [comment]
Closing in favor of #875 .
Wheelchair ramps are often/usually/sometimes and should be mapped seperately which would be difficult to exclude (because it is not even properly defined what/how this is going to be tagged in the wiki). All other ramps are too rare to ask about them specifically, IMO.
